cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/DefaultApplicationSpecification.java

```diff
@@ -93,6 +95,11 @@ public String getName() {
   }
 
   @Override
+  public String getVersion() {
```

Comment:
Seems like version is `nullable`? What happen if the jar manifest doesn't have `Bundle-Version`?